processing/src/main/java/org/apache/carbondata/processing/store/TablePageEncoder.java

```diff
@@ -149,29 +151,26 @@ private void encodeAndCompressDimensions(TablePage tablePage, EncodedData encode
       switch (dimensionSpec.getType(i)) {
         case GLOBAL_DICTIONARY:
           // dictionary dimension
-          indexStorages[indexStorageOffset] =
-              encodeAndCompressDictDimension(
-                  tablePage.getDictDimensionPage()[++dictionaryColumnCount].getByteArrayPage(),
-                  isSortColumn,
-                  isUseInvertedIndex[i] & isSortColumn);
+          indexStorages[indexStorageOffset] = encodeAndCompressDictDimension(
+              tablePage.getDictDimensionPage()[++dictionaryColumnCount].getByteArrayPage(),
+              isSortColumn, isUseInvertedIndex[i] & isSortColumn,
+              CarbonDataProcessorUtil.isRleApplicableForColumn(DimensionType.GLOBAL_DICTIONARY));
           flattened = ByteUtil.flatten(indexStorages[indexStorageOffset].getDataPage());
           break;
         case DIRECT_DICTIONARY:
           // timestamp and date column
-          indexStorages[indexStorageOffset] =
-              encodeAndCompressDirectDictDimension(
-                  tablePage.getDictDimensionPage()[++dictionaryColumnCount].getByteArrayPage(),
-                  isSortColumn,
-                  isUseInvertedIndex[i] & isSortColumn);
+          indexStorages[indexStorageOffset] = encodeAndCompressDirectDictDimension(
+              tablePage.getDictDimensionPage()[++dictionaryColumnCount].getByteArrayPage(),
+              isSortColumn, isUseInvertedIndex[i] & isSortColumn,
+              CarbonDataProcessorUtil.isRleApplicableForColumn(DimensionType.DIRECT_DICTIONARY));
```

Comment:
pass `dimensionSpec.getType(i)` to `CarbonDataProcessorUtil.isRleApplicableForColumn`